messaging/src/test/java/org/axonframework/messaging/responsetypes/AbstractResponseTypeTest.java

```diff
@@ -44,21 +52,21 @@ protected AbstractResponseTypeTest(ResponseType<R> testSubject) {
 
     /**
      * Helper function to make testing of the
-     * {@link ResponseType#matches(Type)} function easier.
+     * {@link ResponseType#matchPriority(Type)} function easier.
      * Takes a {@code methodNameToTest} which it uses to pull a {@link java.lang.reflect.Method} from this abstract
      * class. There after it will pull the return {@link java.lang.reflect.Type} from that method, which it will use as
-     * input for the test subject's match function.
+     * input for the test subject's matchPriority function.
      *
      * @param methodNameToTest a {@link java.lang.String} representing the function you want to extract a return type
      *                         from
-     * @param expectedResult   a {@link java.lang.Boolean} which is the expected result of the matches call
+     * @param expectedResult   a {@link java.lang.Integer} which is the expected result of the matchPriority call
      * @throws NoSuchMethodException if no {@link java.lang.reflect.Method} can be found for the given
      *                               {@code methodNameToTest}
      */
-    protected void testMatches(String methodNameToTest, Boolean expectedResult) throws NoSuchMethodException {
+    protected void testMatchPriority(String methodNameToTest, Integer expectedResult) throws NoSuchMethodException {
```

Comment:
Isn't there still merit in the `testMatches`? 
What if there'd be an implementation of the `matchPriority` that doesn't invoke `match` at all? 
You'd lose the validation of the `match` method in that case.

I understand the desire to minimize the number of tests, but I don't think this is completely ideal.
To not over encumber the test classes, I'd keep the same amount of test methods, but just validate both `match` and `matchPriority` in one test function.
Again, not 100% ideal, but I feel it's the worse of two evils.

What do you think, @Morlack?

Reply:
Or better, invoke both methods in this utility function.

Reply:
I'm on board with the 'invoking both' option

Reply:
:+1: :-)